### ISSUE TITLE
Strange integer(precision) annotation in Blocks.Math.MultiSwitch

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#1664]
**Reported by stefanv on 20 Feb 2015 21:46 UTC**
The aforementioned block contains, in its class annotation,

```mo
...
            textString=DynamicSelect(" ", String(
                  y,
                  minimumLength=1,
                  significantDigits=integer(precision)))),
...
```

The identifier `precision` refers to an Integer parameter of the block, but why is it wrapped with `integer()`?

This is technically illegal (the definition of `integer(x)` in 3.7.1 says, "The argument shall have type Real.", and at the very least, unnecessary.

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/1664

**Comment by modelica-trac-importer** [comment]
**Comment by stefanv on 20 Feb 2015 21:46 UTC**
The same construct also appears in StateGraph.mo.

**Comment by modelica-trac-importer** [comment]
**Comment by stefanv on 6 Mar 2015 16:25 UTC**
[Invalid observation deleted]

**Comment by modelica-trac-importer** [comment]
**Modified by stefanv on 6 Mar 2015 16:25 UTC**

**Comment by modelica-trac-importer** [comment]
**Modified by stefanv on 11 Mar 2015 13:50 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by beutlich on 11 Mar 2015 15:44 UTC**
Stefan, does it work with significantDigits=precision for you? If yes, feel free to commit your changes in maint. branch and trunk.

**Comment by modelica-trac-importer** [comment]
**Comment by stefanv on 11 Mar 2015 15:59 UTC**
I'm not set up to be able to change the MSL.

**Comment by modelica-trac-importer** [comment]
**Comment by beutlich on 11 Mar 2015 16:07 UTC**
If you like you are only two commands away: svn checkout and svn commit

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 22 Jun 2015 13:24 UTC**
Fixed in bdb195f6fe9c1bb3ddcae326bc6846b693d5e93b for maintenance and in f45e6f1b1c73e2e35b092762dd7b0c243d499ced for trunk:

Replaced wrong "integer(precision)" by "precision" in:
- Blocks.Math.MultiSwitch
- StateGraph.Temporary.NumericValue

and correct unsuitable diagram size in Modelica.StateGraph.Examples.ExecutionPaths

**Comment by modelica-trac-importer** [comment]
**Changelog modified by otter on 22 Jun 2015 13:24 UTC**
Replaced wrong "integer(precision)" by "precision" in annotations of blocks Blocks.Math.MultiSwitch and StateGraph.Temporary.NumericValue
